generate/templates/ping360.py.in

```diff
@@ -151,8 +151,27 @@ if __name__ == "__main__":
         p.transmitAngle(x)
     tend_s = time.time()
 
+    print(p)
+
     print("full scan in %dms, %dHz" % (1000*(tend_s - tstart_s), 400/(tend_s - tstart_s)))
 
-    print(p)
+    p.control_auto_transmit(0,399,1,0)
+
+    tstart_s = time.time()
+    for x in range(400):
+        p.wait_message([definitions.PING360_DEVICE_DATA])
+    tend_s = time.time()
+
+    print("full scan in %dms, %dHz" % (1000*(tend_s - tstart_s), 400/(tend_s - tstart_s)))
```

Comment:
The time that you are getting is not the scan time here, it's the time for wait_message to handle all messages.

Reply:
This is the scan time. wait_message is synchronous.